OSDOCS#8126: cert-manager 1.13 changes (#69800)
openshift/openshift-docs: security/cert_manager_operator/cert-manager-operator-release-notes.adoc

For more information, see xref:../../security/cert_manager_operator/index.adoc#cert-manager-operator-about[About the {cert-manager-operator}].

[id="cert-manager-operator-release-notes-1.13"]
== Release notes for {cert-manager-operator} 1.13.0

> Issued: 2024-xx-xx

Author: yes, as mentioned in the "**Points to note**" in the PR description, I will be updating the day before release since the dates for CM are generally prone to changes.

> The following advisory is available for the {cert-manager-operator} 1.13:

Reviewer: Operator version is `1.13.0`. Please consider changing it for all occurrences in the text.

* link:https://access.redhat.com/errata/RHSA-2023:6269-02[RHSA-2023:6269-02]

> Version `1.13` of the {cert-manager-operator} is based on the upstream cert-manager version `v1.13.4`. For more information, see the link:https://cert-manager.io/docs/release-notes/release-notes-1.12/#v1125[cert-manager project release notes for v1.13].

Reviewer: Upstream version will be `v1.13.3` instead of `v1.13.4`.

Reviewer 2: Yes, and need to change the link to `https://cert-manager.io/docs/releases/release-notes/release-notes-1.13#v1133`

[id="cert-manager-operator-new-features-1.13"]
=== New features and enhancements

> * You can now manage certificates for API Server and Ingress Controller by using the {cert-manager-operator}. For more information, see 

Reviewer: ```suggestion
* You can now manage certificates for API Server and Ingress Controller (OpenShift Routes and Ingress) by using the {cert-manager-operator}. For more information, see 
```
Also, might need to conclude the sentence `For more information, see`.

Reviewer: Guess you could link to sections, in `For more information, see`:
- https://70018--ocpdocs-pr.netlify.app/openshift-enterprise/latest/security/cert_manager_operator/cert-manager-creating-certificate#cert-manager-certificate-api-server_cert-manager-creating-certificate
- https://70018--ocpdocs-pr.netlify.app/openshift-enterprise/latest/security/cert_manager_operator/cert-manager-creating-certificate#cert-manager-certificate-ingress_cert-manager-creating-certificate

Author: I have already added this to the notes in PR description. I won't be able to link unless the other change is merged since it will be a new page.

> * You can now manage certificates for xref:../../security/certificates/api-server.adoc#api-server-certificates[API Servers] and xref:../../security/certificates/replacing-default-ingress-certificate.adoc#replacing-default-ingress[Ingress] by using the {cert-manager-operator}.

Reviewer 2: ```suggestion
* You can now manage certificates for xref:../../security/certificates/api-server.adoc#api-server-certificates[API Server] and xref:../../security/certificates/replacing-default-ingress-certificate.adoc#replacing-default-ingress[Ingress] by using the {cert-manager-operator}.
```
Maybe `API Server` is more accurate.

> * With this release, the scope of {cert-manager-operator}, which was previously limited to the AMD64 platform, has now been expanded to include support for managing certificates on IBM Z, IBM Power and ARM64 platforms.

Reviewer: ```suggestion
* With this release, the scope of {cert-manager-operator}, which was previously limited to the OpenShift clusters on amd64 architecture, has now been expanded to include support for managing certificates on OpenShift clusters running on IBM Z (s390x), IBM Power (ppc64le) and arm64 architectures.
```

> * With the latest release, the scope of {cert-manager-operator}, which was previously limited to the AMD64 platform, has now been expanded to include support for managing certificates on IBM Z/Power and ARM64 platforms.

Reviewer: I think RH docs prefer, `amd64` / `x86_64` (I was referring to https://docs.openshift.com/container-platform/4.13/cli_reference/tkn_cli/installing-tkn.html#installing-tkn-on-linux, but please double check if this is correct).

Also, would be good to add the short names for the other platforms included maybe:
- `arm64`
- `ppc64le`
- `s390x`

Author: Since the context here to name the platform and not a code, per our [guidelines](https://redhat-documentation.github.io/supplementary-style-guide/#AMD64), it has to be uppercase.

> * With this release, you can use DNS over HTTPS (DoH) for performing the self-checks during the ACME DNS01 verification. The DNS self-check method can be controlled by using the command line flags, `--dns01-recursive-nameservers-only` and `--dns01-recursive-nameservers`. You can use DoH when all the traffic must be HTTP or HTTPS traffic, for example, when using `HTTPS_PROXY`.

Reviewer: ```suggestion
* With this release, you can use DNS over HTTPS (DoH) for performing the self-checks during the ACME DNS-01 challenge verification. The DNS self-check method can be controlled by using the command line flags, `--dns01-recursive-nameservers-only` and `--dns01-recursive-nameservers`.
```
We can also use DoH without HTTP / HTTPS proxy setup on the cluster.

Author: ok. This was something mentioned on the CM release note.

For more information, see xref:../../security/cert_manager_operator/cert-manager-customizing-api-fields.html#cert-manager-override-arguments_cert-manager-customizing-api-fields[Customizing cert-manager by overriding arguments from the cert-manager Operator API].

[id="cert-manager-operator-1.13-CVEs"]
=== CVEs

> * link:https://access.redhat.com/security/cve/CVE-2023-39615[CVE-2023-39615]

Reviewer: \+ CVEs:
- CVE-2023-3978
- CVE-2023-37788
- CVE-2023-29406

> * link:https://access.redhat.com/security/cve/CVE-[CVE]

Reviewer: Will re-review here once a list of CVEs are added.

Author: sure, thanks! I will notify you for a last scan before I go ahead with merging.

For more information, see xref:../../security/cert_manager_operator/index.adoc#cert-manager-operator-about[About the {cert-manager-operator}].

[id="cert-manager-operator-release-notes-1.13"]
== Release notes for {cert-manager-operator} 1.13.0

Issued: 2024-01-16

The following advisory is available for the {cert-manager-operator} 1.13.0:

* link:https://access.redhat.com/errata/xxxx[xxxx]

Version `1.13.0` of the {cert-manager-operator} is based on the upstream cert-manager version `v1.13.3`. For more information, see the link:https://cert-manager.io/docs/release-notes/release-notes-1.13/#v1133[cert-manager project release notes for v1.13.0].

[id="cert-manager-operator-new-features-1.13"]
=== New features and enhancements

> * You can now manage certificates for API Server and Ingress Controller by using the {cert-manager-operator}. 

Reviewer 3: ```suggestion
* You can now manage certificates for API Server and `IngressController` object by using the {cert-manager-operator}. 
```

Reviewer 3: https://redhat-documentation.github.io/supplementary-style-guide/#ingress

Author: I am using this as a noun as per this
https://redhat-documentation.github.io/supplementary-style-guide/#ingress-controller

//For more information, see xref:../../security/cert_manager_operator/cert-manager-creating-certificate.adoc#cert-manager-creating-certificate

> * With this release, the scope of the {cert-manager-operator}, which was previously limited to the {product-title} on AMD64 architectures, has now been expanded to include support for managing certificates on {product-title} running on IBM Z (`s390x`), IBM Power (`ppc64le`) and ARM64 architectures.

Reviewer 4: ```suggestion
* With this release, the scope of the {cert-manager-operator}, which was previously limited to the {product-title} on AMD64 architecture, has now been expanded to include support for managing certificates on {product-title} running on {ibm-z-name} (`s390x`), {ibm-power-name} (`ppc64le`), and ARM64 architectures.
```

> * With this release, the scope of the {cert-manager-operator}, which was previously limited to the {product-title} on AMD64 architecture, has now been expanded to include support for managing certificates on {product-title} running on {ibm-z-name} (`s390x`), {ibm-power-name} (`ppc64le`) and ARM64 architectures.

Reviewer 3: ```suggestion
* With this release, the scope of the {cert-manager-operator}, which was previously limited to the {product-title} on `amd64` architecture, has now been expanded to include support for managing certificates on {product-title} running on {ibm-z-name} (`s390x`), {ibm-power-name} (`ppc64le`) and `arm64` architectures.
```

Reviewer 3: https://redhat-documentation.github.io/supplementary-style-guide/#bit-64-arm

Author: Using this as a noun as per this:
https://redhat-documentation.github.io/supplementary-style-guide/#arm64
